core/block/template/service.go

```diff
@@ -1,5 +1,322 @@
+	"github.com/anyproto/anytype-heart/core/block/editor/template"
+	"github.com/anyproto/anytype-heart/core/block/export"
+	"github.com/anyproto/anytype-heart/core/block/getblock"
+	"github.com/anyproto/anytype-heart/core/block/object/idresolver"
+	"github.com/anyproto/anytype-heart/core/block/object/objectcreator"
+	"github.com/anyproto/anytype-heart/core/domain"
+	"github.com/anyproto/anytype-heart/pb"
+	"github.com/anyproto/anytype-heart/pkg/lib/bundle"
+	coresb "github.com/anyproto/anytype-heart/pkg/lib/core/smartblock"
+	"github.com/anyproto/anytype-heart/pkg/lib/database"
+	"github.com/anyproto/anytype-heart/pkg/lib/localstore/addr"
+	"github.com/anyproto/anytype-heart/pkg/lib/localstore/objectstore"
+	"github.com/anyproto/anytype-heart/pkg/lib/logging"
+	"github.com/anyproto/anytype-heart/pkg/lib/pb/model"
+	"github.com/anyproto/anytype-heart/space"
+	"github.com/anyproto/anytype-heart/util/internalflag"
+	"github.com/anyproto/anytype-heart/util/pbtypes"
+	"github.com/anyproto/anytype-heart/util/slice"
+)
+
+const (
+	CName           = "template"
+	BlankTemplateID = "blank"
+)
+
+var log = logging.Logger("anytype-mw-api")
+
+type Service interface {
+	StateFromTemplate(templateID, name string) (st *state.State, err error)
+	ObjectApplyTemplate(contextID string, templateID string) error
+	TemplateCreateFromObject(ctx context.Context, id string) (templateID string, err error)
+
+	TemplateCloneInSpace(space space.Space, id string) (templateID string, err error)
+	TemplateClone(spaceID string, id string) (templateID string, err error)
+
+	TemplateExportAll(ctx context.Context, path string) (string, error)
+
+	app.Component
+}
+
+type service struct {
+	picker       getblock.ObjectGetter
+	store        objectstore.ObjectStore
+	spaceService space.Service
+	creator      objectcreator.Service
+	resolver     idresolver.Resolver
+	exporter     export.Export
+}
+
+func New() Service {
+	return &service{}
+}
+
+func (s *service) Name() (name string) {
+	return CName
+}
+
+func (s *service) Init(a *app.App) error {
+	s.picker = app.MustComponent[getblock.ObjectGetter](a)
+	s.store = app.MustComponent[objectstore.ObjectStore](a)
+	s.spaceService = app.MustComponent[space.Service](a)
+	s.creator = app.MustComponent[objectcreator.Service](a)
+	s.resolver = a.MustComponent(idresolver.CName).(idresolver.Resolver)
+	s.exporter = a.MustComponent(export.CName).(export.Export)
+	return nil
+}
+
+// StateFromTemplate creates clone of template object state with empty localDetails and updated objectTypes.
+// Blank template is created in case template object is not found or blank/empty templateID is provided
+func (s *service) StateFromTemplate(templateID, name string) (st *state.State, err error) {
+	if templateID == BlankTemplateID || templateID == "" {
+		return s.blankTemplateState(), nil
+	}
+	if err = getblock.Do(s.picker, templateID, func(b smartblock.SmartBlock) error {
+		if tmpl, ok := b.(*editor.Template); ok {
```

Comment:
you can check that object has template type key in its ObjectTypeKeys array

Reply:
Done!